ADAL/src/ADOAuth2Constants.m

```diff
@@ -68,6 +68,8 @@
 NSString *const BROKER_RESPONSE_KEY             = @"response";
 NSString *const BROKER_HASH_KEY                 = @"hash";
 
+NSString *const ADAL_CLIENT_TELEMETRY           = @"x-ms-clitelem";
```

Comment:
Is there a character limit we're running into here? A longer header key might be preferred here.

Reply:
That's the name STS team came up with, I believe they followed their own naming conventions

Reply:
If this hasn't gone through I'd recommend pushing back, it looks... bad.

Reply:
It has been already implemented and available in testslice

Reply:
I echo @RPangrle here. It does look pretty bad. Just because sts implemented it, does not mean that we cannot change it. We did not apply same argument for changing cloud_instance_name, right?

Reply:
Well, cloud_instance_name was way more dangerous because of hardcoding need, here it's just about the name.... I'll definitely start a thread, but I'm not sure if we have enough evidence to ask to change it....